src/mrack/outputs/pytest_mh.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2023 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""pytest-mh configuration file output module."""
+
+import logging
+
+from mrack.outputs.utils import get_external_id
+from mrack.utils import get_fqdn, is_windows_host, save_yaml
+
+DEFAULT_MHCFG_PATH = "pytest-mh.yaml"
+
+logger = logging.getLogger(__name__)
+
+
+class PytestMhOutput:
+    """
+    Create a configuration file for pytest-mh.
+
+    From job metadata and provisioned information (DB).
+
+    Structure of mh config file is almost the same as our topology in job
+    metadata definition.
+    """
+
+    def __init__(self, config, db, metadata, path=None):
+        """Init the output module."""
+        self._config = config
+        self._db = db
+        self._metadata = metadata
+        self._path = path or DEFAULT_MHCFG_PATH
+
+    def create_mh_config(self):
+        """
+        Create configuration file for python-pytest-multihost.
+
+        Return in form of dict.
+        """
+        cfg = {"domains": []}
+        for domain in self._metadata.get("domains", []):
+            cfgdom = {"type": domain["type"], "hosts": []}
+            cfg["domains"].append(cfgdom)
+
+            for host in domain.get("hosts", []):
+                if "role" not in host:
+                    continue
+
+                provisioned_host = self._db.hosts.get(host["name"], None)
+                if not provisioned_host:
+                    logger.error(f"Host {host['name']} not found in the database.")
+                    continue
+
+                cfgdom["hosts"].append(
+                    {
+                        "hostname": get_fqdn(host["name"], domain.get("name", "")),
+                        "os": "linux" if not is_windows_host(host) else "windows",
```

Comment:
IMHO host_type is semantically incorrect and it is not present in metadata unless explicitly added, this attribute is used only by pytest_multihost. Mrack uses os_type inside is_windows_host but it is also not defined implicitly. We can name it os_type if you prefer.

Reply:
Given that mrack already has the  `os_type` as one of the methods for figuring out if it is windows or not. Then propagating this else where makes sense to me. E.g. we can also, in different PR, enhance pytest-multihost output to translate `os_type` to `host_type` (but still support `host_type`) so that usage of `os_type` could be common for this kind of behavior for all the outputs.

Reply:
Do you set os_type somewhere outside metadata? It is not set in my case so either I can enforce it being set or I need to stick with is_windows_host() anyway.

Reply:
Atm no. 

But it makes sense to me to add it ansible inventory host vars.  Might be useful for some decisions in playbooks.

Something in spirit of:
```
# utils.py
def get_os_type(meta_host).
    os_type = meta_host.get("os_type" )
    if not os_type:
        "os_type" =  "linux" if not is_windows_host(meta_host) else "windows",   

# ansible inventory output
inventory_host["meta_os_type"] = get_os_type(host)
```

Reply:
Ok, I added get_os_type to mrack and renamed python-mh config option to "os_family" to make it clear what it means.

Reply:
Hopefully, this comment is now resolved.

Reply:
LGTM.